src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2FilteringEngine.java

```diff
@@ -27,23 +28,24 @@
 
     private M2FiltersArgumentCollection MTFAC;
     private final double contamination;
+    private final String tumorSample;
 
 
-    public Mutect2FilteringEngine(final M2FiltersArgumentCollection MTFAC) {
+    public Mutect2FilteringEngine(final M2FiltersArgumentCollection MTFAC, final String tumorSample) {
         this.MTFAC = MTFAC;
         contamination = MTFAC.contaminationTable == null ? 0.0 : ContaminationRecord.readContaminationTable(MTFAC.contaminationTable).get(0).getContamination();
-
+        this.tumorSample = tumorSample;
     }
 
     // very naive M1-style contamination filter -- remove calls with AF less than the contamination fraction
     private void applyContaminationFilter(final M2FiltersArgumentCollection MTFAC, final VariantContext vc, final Collection<String> filters) {
-        if (vc.hasAttribute(VCFConstants.ALLELE_FREQUENCY_KEY)) {
-            final double[] alleleFrequencies = getArrayAttribute(vc, VCFConstants.ALLELE_FREQUENCY_KEY);
-            final double maxFrequency = MathUtils.arrayMax(alleleFrequencies);
+        final Genotype tumorGenotype = vc.getGenotypes(tumorSample).get(0);
+            final double[] alleleFractions = GATKProtectedVariantContextUtils.getAttributeAsDoubleArray(tumorGenotype, VCFConstants.ALLELE_FREQUENCY_KEY,
```

Comment:
Instead of `VCFConstants.ALLELE_FREQUENCY_KEY` we want to get `DEPTH_KEY` and `GENOTYPE_ALLELE_DEPTHS` and compute the allele fraction. I _think_ the alleles are not sorted though (we only sort the tumor lods in VCF; we should fix it). Besides I don't think getting allele fractions directly from the genotype fields don't play well with multi-allelic sites. So I suggest we annotate each variant context with a new INFO field `TUMOR_ALT_AF` of type `double[]` when we compute them in `SomaticGenotypingEngine`. I was thinking of taking this approach for `Concordance.java` when I need to retrieve the tumor alt allele fraction of a variant.